### PR TITLE
1693 cross pod task tracking

### PR DESCRIPTION
**Pull request by adberglund**
## Summary
This stores information about a source's running tasks in a cache (currently Redis). Tasks are tracked and invalidated on a per-host level. The list of running tasks is consolidated across hosts  when determining whether to run download/process for a particular source/billing period. 

We only ever want to be running a single task for a source and billing period at one time. So the cache stores a value `{provider_uuid}:{billing_date}`. This covers the two following scenarios:
1) A manifest is not finished processing on one host and another host picks up a task to download/process this manifest. The second host will now be blocked.
2) One host is processing a manifest for a source, and another host picks up a newer manifest for that source. This will also be blocked; however, it should be picked up and run by the next scheduled task that is triggered after the first host is finished processing. 

Again the rule is we should only ever be processing a single manifest for a billing period for a source. 

## Testing

1) Modified docker-compose file to spin up 2 koku-workers (`koku-worker-0` and `koku-worker-1`). 
2) Added `time.sleep(600)` in the `get_report_files` task right before calling `_process_report_file()`
3) Triggered download API on a single AWS source. 
4) Confirmed task started and task added to cache

`koku-worker-0`
```
[2020-02-14 14:25:19,286] INFO 737e2ecd-0203-405b-adf2-0a139358fa9e Added 27505580-d135-4582-8108-aa765e3fea6a:2020-02-01 00:00:00 to cache.
```

5) Triggered download API again
6) Confirmed task was blocked on 2nd worker because it was found in the cache. 

`koku-worker-1`
```
[2020-02-14 14:27:24,275] INFO 37b4a6a8-4c1f-4d18-ad39-20948fe1c9fd 27505580-d135-4582-8108-aa765e3fea6a:2020-02-01 00:00:00 is currently running.
[2020-02-14 14:27:24,280] INFO 37b4a6a8-4c1f-4d18-ad39-20948fe1c9fd Deleted manifest file at /testing/pvc_dir/processing/acct10001/aws/cost-usage-bucket/koku-Manifest.json
[2020-02-14 14:27:24,301] INFO 37b4a6a8-4c1f-4d18-ad39-20948fe1c9fd This manifest has already been downloaded and processed:
 schema_name: acct10001,
 provider_uuid: 27505580-d135-4582-8108-aa765e3fea6a,
 manifest_id:
```

Smokes tests: 

```
== 5627 passed, 128 skipped, 3094 deselected, 9 warnings in 1115.00s (0:18:35) ==
```

### CODE REVIEW

File: koku/masu/test/processor/test_worker_cache.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+class WorkerCacheTest(MasuTestCase):
+    """Test class for the worker cache."""
+
+    def setUp(self):
+        """Set up the test."""
+        super().setUp()
+        cache.clear()
+
+    def tearDown(self):
+        """Set up the test."""
```
Comment (abaiken) [inline koku/masu/test/processor/test_worker_cache.py]: Maybe update this docstring to reflect tear down.

File: koku/masu/test/processor/test_worker_cache.py [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+    def test_task_is_running_true(self):
+        """Test that a task is running."""
+        task_list = [1, 2, 3]
+        _cache = WorkerCache()
+        _cache.set_host_specific_task_list(task_list)
+
+        self.assertTrue(_cache.task_is_running(1))
+
+    def test_task_is_running_false(self):
+        """Test that a task is running."""
```
Comment (abaiken) [inline koku/masu/test/processor/test_worker_cache.py]: update this docstring to reflect that a task is not running - I think that's what this tests but I could be wrong 😅

File: koku/masu/test/processor/test_worker_cache.py (hunk 1 above)
Comment (adberglund) [inline koku/masu/test/processor/test_worker_cache.py]:
```suggestion
        """Tear down the test."""
```

File: koku/masu/test/processor/test_worker_cache.py [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
```
Comment (adberglund) [inline koku/masu/test/processor/test_worker_cache.py]:
```suggestion
        """Test that a task is not running."""
```

**Pull request merged**